tests/test_validate_call.py

```diff
@@ -369,8 +367,8 @@ def foo(self, a: int, b: int):
 
     # insert_assert(exc_info.value.errors(include_url=False))
     assert exc_info.value.errors(include_url=False) == [
-        {'type': 'missing_argument', 'loc': ('a',), 'msg': 'Missing required argument', 'input': ArgsKwargs(())},
-        {'type': 'missing_argument', 'loc': ('b',), 'msg': 'Missing required argument', 'input': ArgsKwargs(())},
+        {'type': 'missing_argument', 'loc': ('a',), 'msg': 'Missing required argument', 'input': ArgsKwargs((x,))},
+        {'type': 'missing_argument', 'loc': ('b',), 'msg': 'Missing required argument', 'input': ArgsKwargs((x,))},
```

Comment:
So the observable change with this approach looks like the error messages will now always show the `self` or `cls` argument. There's debate to be had whether this is a good thing; showing this argument is less implicit but now it doesn't match exactly what the user typed as input arguments, which has potential for confusion.

Reply:
IMO Python already forces users to get used to the idea of the extra self argument with error messages like this:

```python
class C:
  def foo(self, x):
    pass

# TypeError: C.foo() takes 2 positional arguments but 3 were given
C().foo(1, 2)
```

I think there may be times when showing the value of `self` (assuming it has a useful `repr`) might make the error message more useful by providing some extra context about the state of things aside from the actual validation error.

But I'm worried there might be cases where users broadly catch validation errors, serialize them to JSON, and return that in an API. I don't know if that's common but it looks like an intended use case. Then I expect that JSON serialization errors may start appearing, or the validation errors seen by clients (who are far removed from the Python code) will look confusing.

Reply:
Right, good points. I think overall we tend to be ok with error messages changing (we generally justify this by saying that we work to improve error messages). So given that Python itself is a bit mixed here, I'm tempted to argue let's merge this patch as it's a fantastic simplification, and then if we want to work again to improve this later we revisit.